orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/ResolveDeploySourceManifestTask.java

```diff
@@ -63,6 +63,10 @@ public TaskResult execute(@Nonnull StageExecution stage) {
   // Spel is not flattening nested List anymore
   // We flatten them because its possible have a List of manifests resolved by spel
   private List<Object> flattenNestedLists(Object manifests) {
+    if (manifests == null) {
+      return null;
```

Comment:
Should this return an empty list rather than null?

Reply:
Prior to the `flattenNestedLists` function being added, it used to set the context field to null (on line 47), so kept that the same. Happy to change if you think it should

Reply:
I'd feel better if it returned `list.of()` rather than null, I need to get this backported to fix some NPEs.
I like collections being empty rather than null for better fluent API comparability

```java
collection.stream().filter(...).map(...).collect(...)
```
vs

```java
Optional.ofNullable(collection).ifPresent(c -> collection.stream().filter(...).map(...).collect(...)).orElseThrow(() -> new RuntimeException("foo"))
```

Reply:
Lets keep null, since that was the previous behavior

Reply:
I like the idea of return a empty list from `flattenNestedLists` method but the only concern is that changing a the value of the `manifest` field from `null` to `emptyList` could cause a side effect in further tasks.